repos/system_upgrade/common/actors/targetuserspacecreator/libraries/userspacegen.py

```diff
@@ -425,36 +447,59 @@ def gather_target_repositories(context, indata):
     if not target_repoids:
-        raise StopActorExecutionError(
-            message='There are no enabled target repositories for the upgrade process to proceed.',
-            details={'hint': (
-                'Ensure your system is correctly registered with the subscription manager and that'
-                ' your current subscription is entitled to install the requested target version {version}.'
-                ' In case the --no-rhsm option (or the LEAPP_NO_RHSM=1 environment variable is set)'
-                ' ensure the custom repository file is provided regarding the documentation with'
-                ' properly defined repositories or in case repositories are already defined'
-                ' in any repofiles under /etc/yum.repos.d/ directory, use the --enablerepo option'
-                ' for leapp. Also make sure "/etc/leapp/files/repomap.csv" file is up-to-date.'
-                ).format(version=api.current_actor().configuration.version.target)
-            }
-        )
+        reporting.create_report([
+            reporting.Title('There are no enabled target repositories'),
+            reporting.Summary(
+                'During the configuration for the target system, no enabled repositories were found.'
+                ' This can happen when a system is not correctly registered with the subscription manager'
+                ' or, when the leapp --no-rhsm option has been used, no custom repositories have been'
+                ' passed on the command line.'
+            ),
+            reporting.Tags([reporting.Tags.REPOSITORY]),
+            reporting.Flags([reporting.Flags.INHIBITOR]),
+            reporting.Remediation(hint=(
+                'Ensure the system is correctly registered with the subscription manager and that'
+                ' the current subscription is entitled to install the requested target version {version}.'
+                ' If you used the --no-rhsm option (or the LEAPP_NO_RHSM=1 environment variable is set),'
+                ' ensure the custom repository file is provided with'
+                ' properly defined repositories and that the --enablerepo option for leapp is set if the'
+                ' repositories are defined in any repofiles under the /etc/yum.repos.d/ directory.'
+                ' For more information on custom repository files, see the documentation.'
+                ' Finally, verify that the "/etc/leapp/files/repomap.csv" file is up-to-date.'
+            ).format(version=api.current_actor().configuration.version.target)),
+            reporting.ExternalLink(
+                # TODO: How to handle different documentation links for each version?
+                url='https://red.ht/preparing-for-upgrade-to-rhel8',
+                title='Preparing for the upgrade'),
+            reporting.RelatedResource("file", "/etc/leapp/files/repomap.csv"),
+            reporting.RelatedResource("file", "/etc/yum.repos.d/")
+        ])
```

Comment:
@vinzenz please update the stuff to mention `repomap.json` insead of csv. I am keeping it on this PR as we now the #714  is giong to be merged and I would like to avoid useless conflicts in this case.